oak_enclave_runtime_support/src/heap.rs

```diff
@@ -16,14 +16,47 @@
 
 use core::{
     alloc::{GlobalAlloc, Layout},
-    ffi::c_void,
     ops::Deref,
     ptr::NonNull,
 };
-use linked_list_allocator::Heap;
 use oak_restricted_kernel_interface::syscalls::{MmapFlags, MmapProtection};
+use rlsf::{FlexSource, FlexTlsf};
 use spinning_top::Spinlock;
 
+struct Source {}
+
+impl Source {
+    // Restricted Kernel deals in 2 MiB pages, so that's what we use to request memory.
+    const PAGE_SIZE: usize = 0x20_0000usize;
+}
+
+unsafe impl FlexSource for Source {
+    unsafe fn alloc(&mut self, min_size: usize) -> Option<NonNull<[u8]>> {
+        // Ensure that we're allocating page-sized chunks of memory.
+        let size = if min_size % Self::PAGE_SIZE != 0 {
```

Comment:
To double check my understanding: you could also:

```
let missing_size_for_alignmet = min_size % Self::PAGE_SIZE;
let size = min_size + missing_size_for_alignment;
```

Reply:
Close, but `missing_size` would have to be `Self::PAGE_SIZE - min_size % Self::PAGE_SIZE`.

...which is actually much simpler so I'll switch to that.

Reply:
Actually no, doing it like that would increase the size by one page if `size` is already page-aligned. We don't want that.